src/store/cart/index.js

```diff
@@ -0,0 +1,25 @@
+const initialState = { cart: [] };
+
+function cartReducer(state = initialState, action) {
+    switch (action.type) {
+        case 'ADD':
+            return {
+                ...state,
+                cart: [...state.cart, {
+                    name: action.payload.name,
+                    price: action.payload.price,
+                }
+                ]
+            }
+        case 'REMOVE':
+            return {
+                ...state,
+                cart: state.cart.filter(product => product.name !== action.payload.name)
+            };
```

Comment:
If we have more than one of the same item, trying to remove one removes all of them. We want to remove only the one we clicked on.